### ISSUE TITLE
Device MPS: -> RuntimeError Placeholder buffer size is not large enough to contain the Tensor storage of size

### ISSUE THREAD

**Issue by andreabac3** [pytorch/pytorch#78042]
### 🐛 Describe the bug

Hello to everyone,
I am trying to use the torch nightly version which supports the M1 drivers.

My machine is the M1 Pro 14'' base version (16GB Ram, 512GB SSD, 8 CPU cores).

I have tried to install the nightly version in two different ways, but I got the same issue.
In both ways, I have created a new conda env.

1) ```conda install pytorch torchvision torchaudio -c pytorch-nightly```
2) ```pip3 install --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cpu```

The error:
```python3
❯ python3 test.py
Some weights of the model checkpoint at bert-base-cased were not used when initializing BertModel: ['cls.seq_relationship.bias', 'cls.predictions.decoder.weight', 'cls.predictions.bias', 'cls.predictions.transform.dense.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.transform.LayerNorm.bias', 'cls.seq_relationship.weight']
- This IS expected if you are initializing BertModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Reusing dataset conll2003 (/Users/andreabacciu/.cache/huggingface/datasets/conll2003/conll2003/1.0.0/63f4ebd1bcb7148b1644497336fd74643d4ce70123334431a3c053b7ee4e96ee)
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 272.84it/s]
  0%|                                                                                                                                                              | 0/108 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/Users/andreabacciu/Downloads/test.py", line 39, in <module>
    outputs = bert(**model_inputs)
  File "/opt/homebrew/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 1009, in forward
    pooled_output = self.pooler(sequence_output) if self.pooler is not None else None
  File "/opt/homebrew/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 637, in forward
    pooled_output = self.dense(first_token_tensor)
  File "/opt/homebrew/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/torch/nn/modules/linear.py", line 114, in forward
    return F.linear(input, self.weight, self.bias)
RuntimeError: Placeholder buffer size (98304) is not large enough to contain the Tensor storage of size 5505024
```

The code:

```python3
import torch
import transformers as tr
from datasets import load_dataset

from time import time
from tqdm import tqdm
from torch.utils.data import DataLoader

bert = tr.AutoModel.from_pretrained("bert-base-cased")
tokenizer = tr.AutoTokenizer.from_pretrained("bert-base-cased")

datasets = load_dataset("conll2003")
test_data = datasets["test"]


def collate(batch):
    return tokenizer(
        [t["tokens"] for t in batch],
        return_tensors="pt",
        padding=True,
        is_split_into_words=True,
    )


dataloader = DataLoader(
    test_data,
    batch_size=32,
    collate_fn=collate,
    num_workers=0,
)

times_mps = []
bert = bert.to("mps")
# group in batches of 32
for model_inputs in tqdm(dataloader):
    model_inputs = {k: v.to("mps") for k, v in model_inputs.items()}
    t0 = time()
    with torch.no_grad():
        outputs = bert(**model_inputs)
    t1 = time()
    times_mps.append(t1 - t0)

times = []
bert = bert.to("cpu")
# group in batches of 32
for model_inputs in tqdm(dataloader):
    model_inputs = {k: v.to("cpu") for k, v in model_inputs.items()}
    t0 = time()
    with torch.no_grad():
        outputs = bert(**model_inputs)
    t1 = time()
    times.append(t1 - t0)

print("Mean time per batch GPU: {:.2f}s".format(sum(times_mps) / len(times_mps)))
print("Mean time per batch CPU: {:.2f}s".format(sum(times) / len(times)))
```

### Versions

Collecting environment information...
PyTorch version: 1.13.0.dev20220521
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.3 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2)
CMake version: version 3.23.0
Libc version: N/A

Python version: 3.10.2 (main, Feb  2 2022, 05:51:25) [Clang 13.0.0 (clang-1300.0.29.3)] (64-bit runtime)
Python platform: macOS-12.3-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.6
[pip3] pytorch-lightning==1.6.0
[pip3] pytorch3d==0.6.2
[pip3] torch==1.13.0.dev20220521
[pip3] torchaudio==0.11.0
[pip3] torchmetrics==0.8.2
[pip3] torchvision==0.12.0
[conda] pytorch                   1.13.0.dev20220521        py3.10_0    pytorch-nightly

cc @malfet @albanD

**Comment by thipokKub** [comment]
I did not experience this on `1.13.0.dev20220522`. Can you update and try again?

**Comment by mayhewsw** [comment]
I had this problem with 1.13.0.dev20220521, but it went away with 1.13.0.dev20220522.

**Comment by mrshenli** [comment]
seems like the problem no longer exist in the latest nightly. I am closing this issue. @andreabac3 please feel free to reopen if the problem persist

**Comment by xsacha** [comment]
I saw this error printed only after another error occured (the model wasn't converted to mps, but the input was).
Just in case you have the same issue.

**Comment by andreabac3** [comment]
Hi,
Yes, I confirm the issue is fixed in updating to the version `1.13.0.dev20220522`.
Thank you for the support @thipokKub , @mayhewsw , @mrshenli.

Thank you also to @xsacha, now the issue is solved on my side and as you can see from the code posted I have moved the model and inputs to the mps device.

Kind regards,
Andrea Bacciu

**Comment by hwaseem04** [comment]
Hi, I am too facing the issue.

System:
Macbook M1 Pro 14'' base model

Relevant packages versions:
torch                         2.0.0.dev20221213
torchmetrics                  0.11.0
torchvision                   0.13.1a0

The error which I am getting: `RuntimeError: Invalid buffer size: 9.75 GB`

For a reason, I am predicting the results using the trained model on the entire test data(test data size ~ 19962, RGB image).

```
model = model.to(device)
data_s = data_s.to(device)

pred = model(data_s)
```

when the third line is executed, the above error is prompted. Is it due to the limitation of my hardware or is it someother issue?

**Comment by Atomic-Germ** [comment]
> Hi, I am too facing the issue.
> 
> System: Macbook M1 Pro 14'' base model
> 
> Relevant packages versions: torch 2.0.0.dev20221213 torchmetrics 0.11.0 torchvision 0.13.1a0
> 
> The error which I am getting: `RuntimeError: Invalid buffer size: 9.75 GB`
> 
> For a reason, I am predicting the results using the trained model on the entire test data(test data size ~ 19962, RGB image).
> 
> Dataset Used - celebA's test data
> 
> ```
> device = torch.device('mps')
> model = model.to(device)
> data_s = data_s.to(device)
> 
> pred = model(data_s)
> ```
> 
> when the third line is executed, the above error is prompted. Is it due to the limitation of my hardware or is it someother issue?

I get the same result with any batch_size other than 1, using mps.

**Comment by psubodiosa** [comment]
I am getting this error also. The minimal example I have found is the following:
'''
import torch
h = 8
x = torch.nn.Linear(2048,2048).to('mps')
y = torch.rand(16,h,2048,2048).to('mps')
print(x(y).size())
'''
h=4, the program returns the intended result.

h=8, I get the error:
/AppleInternal/Library/BuildRoots/8d3bda53-8d9c-11ec-abd7-fa6a1964e34e/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:705: failed assertion `[MPSTemporaryNDArray initWithDevice:descriptor:] Error: product of dimension sizes > 2**31'

h=16, I get the reported error:
RuntimeError: Invalid buffer size: 4.00 GB

Scaling up, I get the same runtime error with appropriately scaled buffer size in the message.
